synapse/handlers/events.py

```diff
@@ -173,8 +173,11 @@ async def get_event(
         if not event:
             return None
 
-        users = await self.store.get_users_in_room(event.room_id)
-        is_peeking = user.to_string() not in users
+        is_user_in_room = await self.store.check_local_user_in_room(
```

Comment:
might be worth updating the doc string to say 'the **local** user requesting the event'.
It's not directly obvious but since you're going to rely on that being true, we should make that very clear.